drivers/adc/adc_demo/adc_demo.c

```diff
@@ -85,6 +112,144 @@ int32_t adc_demo_remove(struct adc_demo_desc *desc)
 	return SUCCESS;
 }
 
+int32_t adc_get_number_of_channels(void* dev)
+* @brief close all channels
+* @param dev - physical instance of an adc device
+* @return SUCCESS in case of success.
+		  *******************************************************************************/
+		  int32_t close_adc_channels(void* dev)
+{
+	struct adc_demo_desc *desc;
+
+	if(!dev)
+		return -ENODEV;
+
+	desc=dev;
+
+	desc->active_ch = 0;
+
+	return SUCCESS;
+}
+
+/***************************************************************************//**
+ * @brief utility function for computing next upcoming channel
+ * @param ch_mask - active channels .
+ * @param last_idx -  previous index.
+ * @param new_idx - upcoming channel index, return param.
+ * @return 1 if there are more channels, 0 if done.
+*******************************************************************************/
+static bool get_next_ch_idx(uint32_t ch_mask, uint32_t last_idx,
+			    uint32_t *new_idx)
+{
+	last_idx++;
+	ch_mask >>= last_idx;
+	if (!ch_mask) {
+		*new_idx = -1;
+		return 0;
+	}
+	while (!(ch_mask & 1)) {
+		last_idx++;
+		ch_mask >>= 1;
+	}
+	*new_idx = last_idx;
+
+	return 1;
+}
+
+/***************************************************************************//**
+ * @brief function for reading samples
+ * @param dev - physical instance of adc device
+ * @param buff - buffer for reading samples
+ * @param samples - number of samples to receive
+ * @return the number of samples.
+*******************************************************************************/
+uint32_t adc_read_samples(void* dev, uint16_t* buff, uint32_t samples)
+{
+	struct adc_demo_desc *desc;
+	uint32_t k = 0;
+	uint32_t ch = -1;
+
+	if(!dev)
+		return -ENODEV;
+
+	desc=dev;
+
+	if(desc->loopback == NULL) {
+		//default sin function
+		for(k = 0 ; k < samples; k++)
+			buff[k] = sine_lut[k%128];
+
+		return samples;
+	}
+
+	for(int i = 0; i < samples; i++) {
+		while(get_next_ch_idx(desc->active_ch,ch,&ch))
+			buff[k++] = desc->loopback[ch][i%DEFAULT_LOCAL_SAMPLES];
+	}
+	return samples;
+}
+
+/**
+ * @brief get attributes for adc.
+ * @param device- Physical instance of a iio_demo_device.
+ * @param buf - Where value is stored.
+ * @param len - Maximum length of value to be stored in buf.
+ * @param channel - Channel properties.
+ * @param attr_id - Attribute ID
+ * @return Length of chars written in buf, or negative value on failure.
```

Comment:
this is a good example of a positive return related to chars written in a buffer.